yjit/src/backend/arm64/mod.rs

```diff
@@ -234,6 +234,20 @@ impl Assembler
             (opnd0, opnd1)
         }
 
+        fn split_less_than_32_cmp(asm: &mut Assembler, opnd0: Opnd) -> Opnd {
+            match opnd0 {
+                Opnd::Reg(_) | Opnd::InsnOut { .. } => {
+                    match opnd0.rm_num_bits() {
+                        8 => asm.and(opnd0.with_num_bits(64).unwrap(), Opnd::UImm(0xff)),
```

Comment:
I think both of these should be `with_num_bits(32)` since we can use the smaller comparison. I don't know if it will necessarily be that much faster, but it could help.

Reply:
Thanks for reviewing this @kddnewton 

I think Kevin is right, but I also think you might need to split and potentially load opnd0 here, because `and` can't accept memory operands.

Reply:
I would love if that were true. We have a mistake in our assembler for 32bit and on arm. It generates invalid machine code. I plan on fixing it. But didn't want to expand the scope more.

Reply:
Right before this is called there is (see in the diff below)

> let opnd0 = split_load_operand(asm, left);

So it felt a bit weird doing it again. I can if that is the preference.

Reply:
I hadn't spotted that. Looks good Jimmy, well-done 👍